src/core/geometry/qgsrectangle.cpp

```diff
@@ -59,6 +60,24 @@ QgsRectangle::QgsRectangle( const QgsRectangle &r )
   mYmax = r.yMaximum();
 }
 
+QgsRectangle QgsRectangle::fromWkt( const QString &wkt )
+{
+  QgsGeometry geom = QgsGeometry::fromWkt( wkt );
+  if ( ! geom.isGeosValid() || geom.isMultipart() )
```

Comment:
Given that isGeosValid is the most expensive check, I'd move this one last (after all the cheap type/size checks)